### PR TITLE
Studio: Feature Complete

### PR DESCRIPTION
**Pull request by Leland-Takamine**
https://user-images.githubusercontent.com/847683/206627521-3a2e48d2-9e82-4109-88f3-c9911f81072e.mp4

### CODE REVIEW

File: maestro-studio/web/src/Examples.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+          <div className="font-bold">
+            Here are some examples of how you can interact with this element:
+          </div>
+          {sections}
+        </>
+      ) : (
+        <div className="bg-red-100 rounded-md p-5">
+          <p>This element does not have any <b>text</b> or <b>resource id</b> associated with it.</p>
+          <br/>
+          <p>The recommended approach to interact with this element is to first <b>assign it a resource id</b> in code.</p>
```
Comment (axelniklasson) [inline maestro-studio/web/src/Examples.tsx:98]: Shall we add a link to docs explaining how to add a resource id here?

**Review by axelniklasson — APPROVED** [review]

**Comment by dmitry-zaitsev** [comment]
@Leland-Takamine `Conditional` has a wrong indent. `file` should have 1 less indent than `visible` (to be at the same level as `when`)

image

**Comment by dmitry-zaitsev** [comment]
Does it also work for iOS?

File: maestro-studio/web/src/Examples.tsx (hunk 1 above)
Comment (Leland-Takamine) [reply maestro-studio/web/src/Examples.tsx:98]: We don't have those docs yet but yes once we do we should add that!

File: maestro-studio/web/src/ElementSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  onElementSelected:  React.Dispatch<React.SetStateAction<UIElement | null>>
+  hoveredElement: UIElement | null
+  onElementHovered: React.Dispatch<React.SetStateAction<UIElement | null>>
+}) => {
+  const [query, setQuery] = useState("");
+
+  const filteredElements = deviceScreen.elements.filter(element => {
+    if (!element.text && !element.resourceId) return false
+    return !query || element.text?.toLowerCase().includes(query.toLowerCase()) || element.resourceId?.toLowerCase().includes(query.toLowerCase())
+  })
```
[the remark this answers is not in the document]
Comment (Leland-Takamine) [reply maestro-studio/web/src/ElementSearch.tsx]: Updated

File: maestro-studio/web/src/ElementSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+  const sortedElements = filteredElements.sort((a, b) => {
+    const aTextPrefixMatch = query && a.text?.toLowerCase().startsWith(query.toLowerCase())
+    const bTextPrefixMatch = query && b.text?.toLowerCase().startsWith(query.toLowerCase())
+    if (aTextPrefixMatch && !bTextPrefixMatch) return -1
+    if (bTextPrefixMatch && !aTextPrefixMatch) return 1
+    return compare(a.text, b.text) || compare(a.resourceId, b.resourceId)
+  })
```
[the remark this answers is not in the document]
Comment (Leland-Takamine) [reply maestro-studio/web/src/ElementSearch.tsx]: Updated

File: maestro-studio/web/src/App.tsx [hunk 4]
```diff
@@ -1,23 +1,13 @@
...
-      const response = await fetch('/api/device-screen')
-      const responseJson: DeviceScreen = await response.json()
-      setDeviceScreen(responseJson)
-    })()
-  }, [])
-  if (!deviceScreen) {
-    return (
-      <div>Loading...</div>
-    )
+  const getDeviceScreen = async () => {
```
[the remark this answers is not in the document]
Comment (Leland-Takamine) [reply maestro-studio/web/src/App.tsx]: Updated

**Comment by Leland-Takamine** [comment]
> Does it also work for iOS?

As discussed offline - yes!

**Pull request merged**